src/main/java/org/broadinstitute/hellbender/tools/exome/ExonDB.java

```diff
@@ -0,0 +1,370 @@
+     * @throws java.lang.IndexOutOfBoundsException if {@code range} contains invalid indices in this exon data-base.
+     */
+    default List<E> exons(final IndexRange range) {
+        return exons(range.from, range.to);
+    }
+
+    /**
+     * Returns index range of exons that overlap a location.
+     * <p>
+     * If no exon overlap the location, the returned range would have size 0 and would indicate
+     * the insert index where such exons would be found if they were part of the data-base.
+     * </p>
+     *
+     * @param location the target location.
+     * @return never {@code null} but perhaps an empty range.
+     */
+    IndexRange indexRange(final GenomeLoc location);
+
+    /**
+     * Adds exons that overlap a particular location into a collection.
+     * <p>
+     * Exons will be added in genome coordinates order
+     * either using the {@link Collection#add} or {@link Collection#addAll} operations.
+     * </p>
+     *
+     * @param overlapRegion the query location.
+     * @param dest          the destination collection. This collection should
+     *                      support {@link Collection#add} and {@link Collection#addAll}
+     *                      operations.
+     * @return 0 or greater, equal to the number of exons found to be have an overlap with {@code location}.
+     * @throws NullPointerException          if either {@code location} or {@code dest} in {@code null}.
+     * @throws UnsupportedOperationException of any other relevant {@link RuntimeException} when this is
+     *                                       throw by {@code dest} when the resulting exons are added to it.
+     */
+    default int exons(final GenomeLoc overlapRegion, final Collection<E> dest) {
+        if (dest == null)
+            throw new NullPointerException("the destination collection cannot be null");
+        final List<E> exons = exons(overlapRegion);
+        dest.addAll(exons);
+        return exons.size();
+    }
+
+    /**
+     * Returns a list of all exons sorted by genomic coordinates.
+     *
+     * @return never {@code null} but perhaps empty if there is no exon in this data-base.
+     */
+    default List<E> exons() {
+        return exons(GenomeLoc.WHOLE_GENOME);
+    }
+
+    /**
+     * Common interface for lambda functions that perform operations per exon.
+     *
+     * @param <E>
+     */
+    interface ExonTask<E> {
+
+        /**
+         * Perform the corresponding task on a exon
+         *
+         * @param index the index of the target exon.
+         * @param exon  the target exon value.
+         */
+        void run(final int index, final E exon);
+    }
+
+    /**
+     * Index range tuple.
+     * <p>
+     * Field {@link #from} indicates the index of the first exon in the range and field {@link #to} is the index
+     * right after the last exon in the range.
+     * </p>
+     */
+    class IndexRange {
```

Comment:
you're right.  But there are many existing classes that do this, eg:
` org.apache.commons.lang3.Range`
`com.google.common.collect.Range`
`org.broadinstitute.hellbender.tools.picard.illumina.parse.Range`

use an existing one, don't write a new one